.github/workflows/build.yml

```diff
@@ -1,30 +1,49 @@
-name: CMake
-
+name:  Build
 on:
-  push:
-    branches: [ main ]
   pull_request:
-    branches: [ main ]
-
-env:
-  # Customize the CMake build type here (Release, Debug, RelWithDebInfo, etc.)
-  BUILD_TYPE: Release
+  push:
+    branches:
+      - main
 
 jobs:
   build:
-    # The CMake configure and build commands are platform agnostic and should work equally
-    # well on Windows or Mac.  You can convert this to a matrix build if you need
-    # cross-platform coverage.
-    # See: https://docs.github.com/en/actions/configuring-and-managing-workflows/configuring-a-workflow#configuring-a-build-matrix
     runs-on: ubuntu-latest
-
-    container: ros:foxy
-
+    container:
+      image: ${{ matrix.docker_image }}
+    strategy:
+      fail-fast: false
+      matrix:
+        ros_distribution:
+          - foxy
+          - galactic
+          - rolling
+        include:
+          # Foxy Fitzroy (May 2020 - May 2023)
+          - docker_image: rostooling/setup-ros-docker:ubuntu-focal-ros-foxy-ros-base-latest
+            ros_distribution: foxy
+            ros_version: 2
+          # Galactic Geochelone (May 2021 - November 2022)
+          - docker_image: rostooling/setup-ros-docker:ubuntu-focal-ros-galactic-ros-base-latest
+            ros_distribution: galactic
+            ros_version: 2
+          # Rolling Ridley  (June 2020 - Present)
+          - docker_image: rostooling/setup-ros-docker:ubuntu-focal-ros-rolling-ros-base-latest
+            ros_distribution: rolling
+            ros_version: 2
     steps:
-    - uses: actions/checkout@v2
-
-    - name: Build
-      run: . /opt/ros/foxy/setup.sh && colcon build
-
-    - name: Test
-      run: . /opt/ros/foxy/setup.sh && colcon test
+      - uses: ros-tooling/action-ros-ci@v0.2
+        with:
+          package-name: osqp_vendor
+          target-ros2-distro: ${{ matrix.ros_distribution }}
+          colcon-defaults: |
+            {
+              "build": {
+                "cmake-args": [
+                  "-DCMAKE_BUILD_TYPE=Release"
+                ]
+              }
+            }
+      - uses: actions/upload-artifact@v2.2.4
+        with:
+          name: colcon-logs
+          path: ros_ws/log
```

Comment:
Don't you need to change the path based on the `rosdistro`? I think it's difficult to understand the correspondence.
https://github.com/tier4/osqp_vendor/actions/runs/1533847111
![image](https://user-images.githubusercontent.com/31987104/144748006-b2f47e50-78a1-4bb9-a52b-88ec0461e71f.png)

Or IMO, we don't need this log since the normal log output is enough readable.
![image](https://user-images.githubusercontent.com/31987104/144748063-f4aae5b5-0ba8-47b3-a384-8e40192b9f1c.png)

Reply:
I removed artifact at https://github.com/tier4/osqp_vendor/pull/9/commits/37e5cc18a0920b784b8732c6afe5c57b4fb17301.